packages/node/src/node.ts

```diff
@@ -90,21 +90,41 @@ export class Node {
       Opcode.OP_SIGN,
       async (message: ProtocolMessage, next: Function, context: Context) => {
         if (!context.commitment) {
+          // TODO: I think this should be inside the machine for all protocols
           throw Error(
             "Reached OP_SIGN middleware without generated commitment."
           );
         }
-        context.signature = this.signer.signDigest(
-          context.commitment.hashToSign()
-        );
+        context.signature = {
+          v: -1,
+          r: HashZero,
+          s: HashZero
+        };
+        // TODO: Replace the above hack with the below
+        // context.signature = this.signer.signDigest(
+        //   context.commitment.hashToSign()
+        // );
         next();
       }
     );
   }
 
   private registerIoMiddleware() {
-    // TODO:
-    this.instructionExecutor.register(Opcode.IO_SEND, () => {});
+    this.instructionExecutor.register(
+      Opcode.IO_SEND,
+      async (message: ProtocolMessage, next: Function, context: Context) => {
+        await this.messagingService.send(context.outbox[0].toAddress, {
```

Comment:
Just to confirm, a protocol is never restarted mid-execution, right?

So there's never a case where this `context.outbox` array is gonna be empty?

Reply:
That is correct